### DISCUSSION TITLE
Git Release Tags and Version Numbering in Access

### DISCUSSION THREAD

**Discussion by NotionalGravitas** [joyfullservice/msaccess-vcs-addin#518] [Q&A] [answered]
Firstly, amazing tool. I have periodically seached for something like this for years and suddenly here it is. 

https://github.com/joyfullservice/msaccess-vcs-addin/issues/410#issuecomment-1626022393
With this comment as a starting point, I have been trying to adapt my extremely clumsy access development workflow to integrate git as well.  

My concern is how to best handle version identifiers in Access when using git. 

- Previously (pre-git) I have had a local configuration table in my access files, which stored a version identifier as a text field, and which I periodically updated when I make changes... when I remembered. This is usually displayed somewhere on a form in the UI.
- I actually prefer what I see being used in this add-in, making a custom database properties field called appVersion. This seems much more reliable.

But regardless to how the version identifier is stored, I'm trying to achieve some correllation between the access file and the git release tags (along the lines of a "git describe --tags"). 

What I was aiming for was finding a way to record the release from which the access file has been built and managing versions this way. 

I've considered writing a module in VBA to run the above 'git describe' command after build and save the result, but I believe that'll require the built file to know where the source folder is and in my current environment the source isn't kept with the binary file and the repo names don't exactly match the binary names. 

I was thinking the environment issues above might be resolved with a bit of housekeeping and maybe some environment variables, but in light of that previous link, I wonder if I'm tackling the whole idea of versioning access files the wrong way.

Has anyone here come up with a good workflow for versioning access while using git and this add-in?

**Comment by hecon5** [comment, marked as answer]
We use a modified version of `basDeploy` found in https://github.com/msaccess-vcs-integration/msaccess-devops. 

We store the version number as an Access file property, it gets stored as an item in the addin code base (see how it's done in the addin). 

While this doesn't immediately integrate with git, it does make it consistent so you could build some scripts to find that line in `dbs-properties.json` under `Items:AppVersion`. I think someone put together an issue that built a runner for this to auto-deploy (build, compile, deploy), but I've not used it; using the devops tools I linked above works swimmingly for us in our workflow.

I've got some updates to that, I'll put those together and push that up today, thanks for the reminder.

**Reply by NotionalGravitas** [reply]
Thanks for the pointers. That sounds like the kind of thing I'm looking for. I'll go have a look around.
